website/src/app/pages/visualization/visualization.component.ts

```diff
@@ -0,0 +1,103 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Options } from 'ngx-vega';
+import { Subscription } from 'rxjs';
+import { createGeoZoomPatch } from 'src/app/shared/components/visualization-page/shared/geomap-zoom-patch';
+
+interface Visualization {
+  title: string;
+  description: string;
+  spec: string;
+  options: Options;
+  content: string;
+  sql: string;
+  csv: string;
+}
+
+interface VisualizationDictionary {
+  [key: string]: Visualization;
+}
+
+@Component({
+  selector: 'agc-visualization',
+  templateUrl: './visualization.component.html',
+  styleUrls: ['./visualization.component.scss']
+})
+export class VisualizationComponent implements OnInit {
+  visualizationID = '';
+  private routeSub: Subscription = {} as Subscription;
+  visualizations: VisualizationDictionary = {
```

Comment:
Move these visualizations into ngxs state and actual config data into its own .ts file (ideally its json, but we need support for creating functions that you can't do in json as it is).

Reply:
Look up routing to see how you can prefetch this data (using a route guard)

Reply:
Look up routing to see how you can prefetch this data (using a route guard)